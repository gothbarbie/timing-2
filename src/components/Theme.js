import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'

import theme from './themes/default'

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

Theme.propTypes = {
  children: PropTypes.node,
}

export default Theme
