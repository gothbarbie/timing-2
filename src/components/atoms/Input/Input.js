import React from 'react'
import styled from 'styled-components'

const InputStyle = styled.input`
  border-radius: 4px;
  border: 1px solid ${({ theme }) => theme.colors.gray};
  padding: 1rem 1.5rem;
  font-size: 1.6rem;
  color: ${({ theme }) => theme.colors.dark};
  outline: none;

  &:focus {
    border: 1px solid ${({ theme }) => theme.colors.primary};
  }
`

const Input = ({ name, onChange, type = 'text' }) => {
  return <InputStyle name={name} onChange={onChange} type={type} />
}

export default Input
