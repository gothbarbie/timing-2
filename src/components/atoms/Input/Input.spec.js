import React from 'react'
import { shallow } from 'enzyme'

import { theme } from '../../Theme'
import Input from './Input'

describe('generic/atoms/Input', () => {
  it('renders', () => {
    const component = shallow(<Input theme={theme} />)

    expect(component).toMatchSnapshot()
  })
})
