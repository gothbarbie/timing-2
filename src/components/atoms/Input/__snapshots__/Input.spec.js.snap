// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic/atoms/Input renders 1`] = `
<styled.input
  type="text"
/>
`;
